### DISCUSSION TITLE
docker重启后，出现2002错误。请问 mysql_autoload.php怎么找到这个文件改强制连接

### DISCUSSION THREAD

**Discussion by 7161457** [MoeNetwork/Tieba-Cloud-Sign#215] [疑问] [answered]
我是群辉nas。
我是docker pull moenetwork/tieba-cloud-sign 拉出来的程序，然后启动后连接自己搭建的mariadb数据库，已经部署成功了，并且正常签到。
由于主机断电导致docker程序重启后，出现连接数据库失败，经测试数据库是正常，我看需要开启 MySQL 连接方式强制功能，要打开 mysql_autoload.php，
请问 mysql_autoload.php怎么找到这个文件。

**Comment by n0099** [comment]
什么叫
> 我看需要开启 MySQL 连接方式强制功能，要打开 mysql_autoload.php，

您从哪儿看的，csdn吗

**Reply by 7161457** [reply]
![1665657340372](https://user-images.githubusercontent.com/113887555/195575338-42076283-e39f-44f9-9f4a-185f70029aad.jpg)
程序页写的出现2002，改这个文件

**Reply by n0099** [reply]
https://github.com/MoeNetwork/Tieba-Cloud-Sign/blob/c0a58bde353e006de88047ec94a1735fb5a4a711/lib/mysql_autoload.php

您说的很对，但这是几百年前（ https://github.com/MoeNetwork/Tieba-Cloud-Sign/commit/7ec2c4c53b19647bf6a0d8bb414a50bd9801cdee ）由tc第一代作者wmzz写的一个选择使用`php-mysql`还是`php-mysqli`提供的api来连接数据库
与此同时php-mysql扩展早已在php7中被正义删除，除非您还在用php5.x不然您换了只会报错不存在函数`mysql_xxxxxxx`
> This extension was deprecated in PHP 5.5.0, and it was removed in PHP 7.0.0. Instead, the [MySQLi](https://www.php.net/manual/en/book.mysqli.php) or [PDO_MySQL](https://www.php.net/manual/en/ref.pdo-mysql.php) extension should be used

**Reply by 7161457** [reply]
你好，你这个连接是程序库的地址，我在这里改应该没用啊，我的主机还是改不了，我的意思是，怎么在我的nas里找到这个文件并且更改他。使我的本地数据生效。

**Reply by 7161457** [reply]
我是群辉自带的docker程序
![1665671086330](https://user-images.githubusercontent.com/113887555/195624046-4b07a989-097d-47af-9307-10f290967fa6.jpg)
  拉出来的镜像然后安装的容器，我并不知道他生成的[mysql_autoload.php]这个文件在什么地方，我的意思是应该通过什么方式找到他。

**Reply by n0099** [reply]
https://stackoverflow.com/questions/30172605/how-do-i-get-into-a-docker-containers-shell

**Reply by n0099** [reply]
我的意思是即便您改了也只会变成另一个错误：
> 不然您换了只会报错不存在函数

**Reply by 7161457** [reply]
你好，请问从我的日志中 能看出来，我的具体是哪里出现问题了吗，并且怎么解决问题？可以让docker重启后，不会出现2002错误。
![1665659534934](https://user-images.githubusercontent.com/113887555/195736546-648630fa-26e7-4385-b31e-5f776a30ba53.jpg)
![1665659534951](https://user-images.githubusercontent.com/113887555/195736553-a9f24e64-46c1-4d29-b337-081c0a279b16.jpg)
![1665659534908](https://user-images.githubusercontent.com/113887555/195736556-267f9e81-5f2f-481d-96a8-5beafcec80e2.jpg)
![1665659534919](https://user-images.githubusercontent.com/113887555/195736558-5f879f8f-f308-4fc9-819e-265def7a5d9e.jpg)

**Reply by n0099** [reply]
在楼下看过了
但我不懂docker

**Reply by 7161457** [reply]
好的 感谢你

**Comment by BANKA2017** [comment, marked as answer]
请检查数据库的状态

**Reply by 7161457** [reply]
docker经过多次重启都是显示2002，且数据库用phpmyadmin连接是正常的，docker日志如下，
![1665659534904](https://user-images.githubusercontent.com/113887555/195582356-496093b4-1dba-45ce-bad2-91b938f1c314.jpg)
![1665659534908](https://user-images.githubusercontent.com/113887555/195582358-159ae236-52d5-46ed-9222-7f9577f0f8f5.jpg)
![1665659534919](https://user-images.githubusercontent.com/113887555/195582364-bf7a7fff-dd38-43c7-8ec2-c2949e999ed2.jpg)
![1665659534934](https://user-images.githubusercontent.com/113887555/195582369-7b381c5d-12d3-41b8-8c31-8649084843a5.jpg)
![1665659534945](https://user-images.githubusercontent.com/113887555/195582372-774b0745-ff3c-4ffe-8cf1-f89c2211bd5a.jpg)
![1665659534951](https://user-images.githubusercontent.com/113887555/195582375-e6c5af5f-2a25-4939-b96b-09c7a342c3b8.jpg)

**Reply by BANKA2017** [reply]
我重新看了一遍上面的描述，目前您使用docker来跑程序本体，然后连接到安装在群晖本身上面的mariadb（而不是[`docker-compose.yml`](https://github.com/MoeNetwork/Tieba-Cloud-Sign/blob/master/docker/docker-compose.yml)里面自带的那个），然后某次断电重启后就连不上数据库了？

**Reply by 7161457** [reply]
是的，是每次断电都是连不上数据库，手动重启也连不上数据库。

**Reply by 7161457** [reply]
[docker-compose.yml](https://github.com/MoeNetwork/Tieba-Cloud-Sign/blob/master/docker/docker-compose.yml)这个我也在ubuntu16里面尝试了，也是不行。
![1665791975347](https://user-images.githubusercontent.com/113887555/195960576-720980c3-ca77-4a56-ae81-ce27fbe9128c.jpg)
![1665791975353](https://user-images.githubusercontent.com/113887555/195960579-ccf36fdb-25a6-40d6-9c4d-14c73503444e.jpg)

**Reply by 7161457** [reply]
还有一个情况就是，我用另外一个人的镜像尝试了，他的是4.92的版本，连接自己数据库，包括docker-compose up -d，重启都是可以的，但是就是用户添加超时，只能手动向数据库添加 bduss。用他的自动升级到5.0也是不行。扫码超时，手动超时。
![1665793372435](https://user-images.githubusercontent.com/113887555/195960783-0ea4948d-8711-4216-8840-f768aac9e665.jpg)

**Reply by n0099** [reply]
还有人用ubuntu1604？

https://github.com/zsnmwy/Tieba-Cloud-Sign
updated 4 years ago

**Reply by 7161457** [reply]
是的 我用的就是

**Reply by n0099** [reply]
![image](https://user-images.githubusercontent.com/13030387/196006828-78050603-bf57-4342-b19c-74c933e52c1a.png)
![image](https://user-images.githubusercontent.com/13030387/196006896-9a33350a-c381-43ed-9f69-38c7af5d73b4.png)

**Reply by 7161457** [reply]
感谢各位朋友凌晨还帮忙调试，群辉是大家没有所以没法测试吗？ubuntu16不行的话，那个版本可以，我装机试下。

**Reply by 7161457** [reply]
还有一个问题，主页里说可以docker-compose up -d，这个是在那个版本里测试的？我可以装这个系统

**Reply by 7161457** [reply]
朋友要用ubuntu16调试的话，我可以把我的几个端口都转发出去，连我的ubuntu16调试

**Reply by BANKA2017** [reply]
我那个应该是vm的问题，跟系统版本无关，你也可以试试装docker compose v2（明显的区别就是v1的命令是docker-compose，v2的命令是 docker compose）

**Reply by 7161457** [reply]
你好 ，我看了我的版本 我的也是V2  docker-compose --version和 docker compose --version的命令我试了一下，都可以用。
![1665887018779](https://user-images.githubusercontent.com/113887555/196014959-fd44dbe0-6f42-43ea-bb03-7c949d35b7f4.jpg)

**Reply by n0099** [reply]
那您试试v2

**Reply by 7161457** [reply]
v2不行  已经用群辉的WebStation搭建，这次好了。

**Reply by n0099** [reply]
![image](https://user-images.githubusercontent.com/13030387/196036380-e9a4c749-08f9-437c-885b-b82ed4939dd9.png)
![image](https://user-images.githubusercontent.com/13030387/196036415-b5a4ac3f-443a-4dc6-9a62-8afe8b069d28.png)
![image](https://user-images.githubusercontent.com/13030387/196036368-08ce8afe-6ae1-497a-9208-7abeb2795471.png)

**Reply by 7161457** [reply]
谢谢，虽然好了 但是我还是在ubuntu 18  ubuntu 20 都下载都用了 也是启动不了。

**Reply by BANKA2017** [reply]
比较尴尬的是，其实我至今还没能够把镜像拉下来，晚点我拉下来看看能不能复现

**Reply by 7161457** [reply]
由于装在webstation 不能执行自动签到计划任务，我又用另外一个签到框架（QianDao for Python3
Qiandao —— 一个HTTP请求定时任务自动执行框架 base on HAR Editor and Tornado Server）加了一个定时任务，这样就可以每天定时了的骚操作。
![1665925846515](https://user-images.githubusercontent.com/113887555/196037764-8329e6bf-8681-4262-a350-4c9d967d8bec.jpg)
![1665925864566](https://user-images.githubusercontent.com/113887555/196037765-efe3c918-b54d-4fd4-8dfd-6c126ac4f63f.jpg)
![1665925931834](https://user-images.githubusercontent.com/113887555/196037766-e0adebb3-73dd-4a49-b6a4-0705e7522ac7.jpg)
![1665925986422](https://user-images.githubusercontent.com/113887555/196037768-9e1d2a0f-8020-4c68-901c-5a150fa8885c.jpg)
![1665926055864](https://user-images.githubusercontent.com/113887555/196037769-07229d2d-a283-468c-be2d-ec778e0f642a.jpg)

**Reply by 7161457** [reply]
> 比较尴尬的是，其实我至今还没能够把镜像拉下来，晚点我拉下来看看能不能复现

安装docker 和compose 用下面方法 
安装docker 
卸载旧版本
sudo apt-get remove docker docker-engine docker.io containerd runc
设置仓库
更新 apt 包索引
sudo apt-get update
安装 apt 依赖包，用于通过HTTPS来获取仓库
sudo apt-get install \
    apt-transport-https \
    ca-certificates \
    curl \
    gnupg-agent \
    software-properties-common
添加 Docker 的官方 GPG 密钥
curl -fsSL https://mirrors.ustc.edu.cn/docker-ce/linux/ubuntu/gpg | sudo apt-key add -

9DC8 5822 9FC7 DD38 854A E2D8 8D81 803C 0EBF CD88 通过搜索指纹的后8个字符，验证您现在是否拥有带有指纹的密钥。
sudo apt-key fingerprint 0EBFCD88
   
pub   rsa4096 2017-02-22 [SCEA]
      9DC8 5822 9FC7 DD38 854A  E2D8 8D81 803C 0EBF CD88
uid           [ unknown] Docker Release (CE deb) <docker@docker.com>
sub   rsa4096 2017-02-22 [S]

使用以下指令设置稳定版仓库
sudo add-apt-repository \
   "deb [arch=amd64] https://mirrors.ustc.edu.cn/docker-ce/linux/ubuntu/ \
  $(lsb_release -cs) \
  stable"

安装 Docker Engine-Community
更新 apt 包索引
sudo apt-get update
安装最新版本的 Docker Engine-Community 和 containerd ，或者转到下一步安装特定版本：
sudo apt-get install docker-ce docker-ce-cli containerd.io

安装 Docker Compose

curl -L https://get.daocloud.io/docker/compose/releases/download/v2.11.2/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
chmod +x /usr/local/bin/docker-compose

**Reply by 7161457** [reply]
> 比较尴尬的是，其实我至今还没能够把镜像拉下来，晚点我拉下来看看能不能复现

另外我的docker不是远古版本的，是比较新的版本哦。
![1665927089781](https://user-images.githubusercontent.com/113887555/196038270-b6d6b6ab-d29b-44a2-acd5-ea27a0402815.jpg)

**Reply by n0099** [reply]
![image](https://user-images.githubusercontent.com/13030387/196045962-26e8a060-15ce-40d3-b691-2b720c9654b5.png)

**Reply by BANKA2017** [reply]
> ![image](https://user-images.githubusercontent.com/13030387/196045962-26e8a060-15ce-40d3-b691-2b720c9654b5.png)

您说的都对，但目前这个坑是我接下来的我会继续跟进，以及不要再用聊天截图污染讨论串了
